### DISCUSSION TITLE
How to disable content interaction when bottom sheet is animating?

### DISCUSSION THREAD

**Discussion by likern** [gorhom/react-native-bottom-sheet#575] [Q&A] [answered]
When bottom sheet starts animation (close animation) it's expected that animation can't be interrupted.
But in reality I cant interrupt animation and put it into inconsistent state (animation is stopped in the middle, while allowed states are `-1` and `full height`)

https://user-images.githubusercontent.com/842306/129241632-974fe973-3808-46de-821f-f71e667aa1b0.mp4

**Comment by gorhom** [comment]
@likern could you provide a reproducible sample code based on v4 template , i am working on a solution for this issue and i want to test it on your use-case

**Comment by likern** [comment]
Yes, I can. But my use-case is really simple. It's reproducible without special code.

Let's say user is pressing button very fast (like he is nervous). On first press bottom sheet starts opening. On second press bottom sheet is interrupted (it gets second press) and stays in the middle of transition.

**Reply by likern** [reply]
https://user-images.githubusercontent.com/842306/132949229-1d43dee0-e6fa-4bf8-a172-0bb6baebc8f9.mp4

**Comment by likern** [comment]
The original answer is the same, just animating from top to bottom.

**Comment by gorhom** [comment]
Okay, I’ll push a new shortly to prevent interaction with sheet content while animating

**Reply by likern** [reply]
I think ideally we should be able to interrupt animation. But it should never stop at the middle.

If we release gesture (and direction / velocity didn't change) we continue animating to target position. If direction was changed (I interrupted and decided to close bottom sheet) we animal to original position.

And I think about adding option to put protection view (invisible) to block interaction with content. So not only bottom sheet will always animate, but that users can't accidentally interact with content of bottom sheet (like selection tag in my case)

**Comment by gorhom** [comment]
@likern could you this pr https://github.com/gorhom/react-native-bottom-sheet/pull/638 , thanks

**Reply by likern** [reply]
Thank you! I will test in a recent days on my project 🤗

**Reply by gorhom** [reply]
@likern any luck ? :)

**Reply by likern** [reply]
Sorry for delay, I will test today))

**Reply by likern** [reply]
@gorhom I've tested and I'm still can interrupt animation in both cases.

**Reply by likern** [reply]
https://user-images.githubusercontent.com/842306/133935180-4186a8ca-bd58-4c46-937c-19296b33209c.mp4

**Comment by likern** [comment]
Should I put `enableContentInteractionWhileAnimating={false}` into `<BottomSheet />` component?

This is how I tested
```typescript
import BottomSheet, {
  BottomSheetView,
  BottomSheetBackdrop,
  useBottomSheetDynamicSnapPoints
} from '@gorhom/bottom-sheet';
...
return (
      <>
        <BottomSheet
          ref={reactNativeBottomSheetRef}
          index={-1}
          snapPoints={animatedSnapPoints}
          handleHeight={animatedHandleHeight}
          contentHeight={animatedContentHeight}
          enablePanDownToClose={true}
          enableOverDrag={false}
          enableContentInteractionWhileAnimating={true}
          handleComponent={Handle}
          backdropComponent={Backdrop}
        >
          <BottomSheetView onLayout={handleContentLayout}>
            {children}
          </BottomSheetView>
        </BottomSheet>
      </>
    );
```

with both `enableContentInteractionWhileAnimating` set to `true` or `false`. Same result.

**Reply by gorhom** [reply]
Hmm that is weird, could you provide a minimal sample code of your use-case ?

**Reply by likern** [reply]
I've provided sample code https://github.com/gorhom/react-native-bottom-sheet/pull/652

**Comment by gorhom** [comment, marked as answer]
Hi @likern, would you mind testing https://github.com/gorhom/react-native-bottom-sheet/pull/769
